tests/test_cases/test_cocotb/common.py

```diff
@@ -29,3 +35,19 @@ class MyException(Exception): ...
 
 
 class MyBaseException(BaseException): ...
+
+
+@contextmanager
+def assert_takes(
+    time: float, unit: TimeUnit, cmp: Callable[[int, int], bool] = operator.eq
+) -> Generator[None, None, None]:
+    """Assert that the code takes at least `time` in `unit` to run"""
+    expected = get_sim_steps(time, unit)
+    start = get_sim_time("step")
+    yield
+    end = get_sim_time("step")
+    actual = end - start
+    actual_in_units = get_time_from_sim_steps(actual, unit)
+    assert cmp(actual, expected), (
+        f"Expected the code to take {time} {unit}, actually took {actual_in_units} {unit}"
```

Comment:
```suggestion
        f"Expected the code to take {time}{unit}, actually took {actual_in_units}{unit}"
```
(not sure how we print units in other places actually...)